### PR TITLE
refactor: Simplify Component.firstChild, .lastChild, and .findParent

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

These methods can be simplified using the extension `Iterable.firstOrNull` getter.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [-] I have updated/added tests for ALL new/updated/fixed functionality.
- [-] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples`.

## Breaking Change

- [-] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame/lib/src/components/component.dart [hunk 1]
```diff
@@ -265,19 +266,16 @@ class Component {
...
-  /// Returns the first child that matches the given type [T].
-  ///
-  /// As opposed to `children.whereType<T>().first`, this method returns null
-  /// instead of a [StateError] when no matching children are found.
+  /// Returns the first child that matches the given type [T], or null if there
+  /// are no such children.
   T? firstChild<T extends Component>() {
-    final it = children.whereType<T>().iterator;
-    return it.moveNext() ? it.current : null;
+    return children.whereType<T>().firstOrNull;
```
Comment (spydon) [inline packages/flame/lib/src/components/component.dart:272]: We could make this more efficient if we check whether there is a query registered for `T`

Comment (st-pasha) [reply packages/flame/lib/src/components/component.dart:272]: That could be on the `ComponentSet`, which would override the `whereType<T>`.

Comment (spydon) [reply packages/flame/lib/src/components/component.dart:272]: True true

**Review by spydon — APPROVED** [review]

File: packages/flame/lib/src/components/component.dart (hunk 1 above)
Comment (spydon) [reply packages/flame/lib/src/components/component.dart:272]: Opened https://github.com/bluefireteam/ordered_set/issues/20

**Pull request merged**